src/autoschedulers/adams2019/cost_model_generator.cpp

```diff
@@ -448,13 +450,12 @@ class CostModel : public Generator<CostModel<training>> {
             // to the model weights.
             Derivative d_loss_d = propagate_adjoints(loss_output);
 
-            Weight *weights[] = {&head1_filter, &head1_bias,
-                                 &head2_filter, &head2_bias,
-                                 &filter1, &bias1};
-
-            for (Weight *w : weights) {
```

Comment:
It's losing the ability to iterate over the weights here that I was referring to.

Reply:
That could be fixed pretty trivially by making `backprop()` a virtual method and adding a `ModelWeightBase` class.

Reply:
(Just pushed a revision to do this.)